### PR TITLE
Hide input component

### LINKED ISSUE #1183: Hide passwords and other credentials in UI
## Describe the bug
Credential inputs in the configuration are shown plain-text.

## Tasks
- [x] Create a component that would hide input by default, but would have an option to show plaintext on click (0.5d) @ilija-lazoroski 
- [x] Convert all of the sensitive inputs to use this component (0.25d) @ilija-lazoroski 
- [ ] Review and address comments (0.5d) @ilija-lazoroski

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do? 

Fixes #1183  

Add any further explanations here. 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing? 
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] ~Was the documentation framework updated to reflect the changes?~

## Testing Checklist

* [x] ~Added relevant unit tests?~
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...} 
* [x] If applicable, add screenshots or log transcripts of the feature working

## Explain Changes

image
image
image

### CODE REVIEW

**Review by shreyamalviya — CHANGES_REQUESTED** [review]
Why is the spacing before the "should be string" more in configuration options with the hidden eye component? Let's make it uniform.

image

File: monkey/monkey_island/cc/ui/src/components/ui-components/HideInput.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      hidden: false
+    };
+    this.toggleShow = this.toggleShow.bind(this);
+  }
+
+  toggleShow() {
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/components/ui-components/HideInput.js:15]: My javascript is rusty, but if you define this like 
```suggestion
  toggleShow = () => {
```
then you can remove the `bind()` on line 12.

File: monkey/monkey_island/cc/ui/src/components/ui-components/HideInput.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    this.setState({hidden: ! this.state.hidden});
+  }
+
+  onChange(e) {
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/components/ui-components/HideInput.js:19]: Again, my javascript is rusty. Does this also need a call to `bind()` or equivalent?

File: monkey/monkey_island/cc/ui/src/styles/components/HideInput.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.eye-button{
+  padding: 5px !important;
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/styles/components/HideInput.scss:2]: Does this really need the `!important`?

Comment (mssalvatore) [reply monkey/monkey_island/cc/ui/src/styles/components/HideInput.scss:2]: I don't see this class being used anywhere.

File: monkey/monkey_island/cc/ui/src/components/ui-components/HideInput.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import {InputGroup, FormControl} from 'react-bootstrap';
+import '../../styles/components/HideInput.scss'
+
+class HideInput extends React.PureComponent {
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/components/ui-components/HideInput.js:5]: Maybe a name like `SensitiveTextInput` would be better? The input isn't always hidden.

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/ui/src/styles/components/HideInput.scss (hunk 3 above)
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/ui/src/styles/components/HideInput.scss:2]: Deleted.

File: monkey/monkey_island/cc/ui/src/components/ui-components/HideInput.js (hunk 2 above)
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/ui/src/components/ui-components/HideInput.js:19]: https://newbedev.com/why-and-when-do-we-need-to-bind-functions-and-eventhandlers-in-react
Apparently there is couple ways of binding, in which I have use the third one. 

`onChange={(event) => this.onChange(event)}`

**Review by mssalvatore — APPROVED** [review]

**Review by mssalvatore — CHANGES_REQUESTED** [review]
The develop branch has been backmerged into this branch. We need to undo/fix that before we merge this branch back into develop.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**